File: package/python/src/fake_chinese_checker.py
def is_fake_chinese(text: str) -> str:
    """
    Check if a string contains only Chinese characters (Kanji).

    Args:
        text: Input string to check

    Returns:
        "正" if all characters are Chinese/Kanji, "誤" otherwise
    """
    if not text:
        return "誤"

    for char in text:
        # Check if character is a CJK ideograph (Chinese/Japanese/Korean unified ideographs)
        if not ("\u4e00" <= char <= "\u9fff"):
            return "誤"

    return "正"

File: package/typescript/src/fakeChineseChecker.ts
/**
 * Check if a string contains only Chinese characters (Kanji).
 *
 * @param text - Input string to check
 * @returns "正" if all characters are Chinese/Kanji, "誤" otherwise
 */
export function isFakeChinese(text: string): string {
  if (!text) {
    return "誤";
  }

  for (const char of text) {
    // Check if character is a CJK ideograph (Chinese/Japanese/Korean unified ideographs)
    const codePoint = char.codePointAt(0);
    if (!codePoint || !(codePoint >= 0x4e00 && codePoint <= 0x9fff)) {
      return "誤";
    }
  }

  return "正";
}
